base/show.jl

```diff
@@ -924,36 +925,38 @@ show_linenumber(io::IO, line, file) = print(io, "#= ", file, ":", line, " =#")
 show_linenumber(io::IO, line, file::Nothing) = show_linenumber(io, line)
 
 # show a block, e g if/for/etc
-function show_block(io::IO, head, args::Vector, body, indent::Int)
+function show_block(io::IO, head, args::Vector, body, indent::Int, quote_level::Int)
     print(io, head)
     if !isempty(args)
         print(io, ' ')
         if head === :elseif
-            show_list(io, args, " ", indent)
+            show_list(io, args, " ", indent, 0, quote_level)
         else
-            show_list(io, args, ", ", indent)
+            show_list(io, args, ", ", indent, 0, quote_level)
         end
     end
 
     ind = head === :module || head === :baremodule ? indent : indent + indent_width
-    exs = is_expr(body, :block) ? body.args : Any[body]
+    exs = (is_expr(body, :block) || is_expr(body, :quote)) ? body.args : Any[body]
     for ex in exs
         print(io, '\n', " "^ind)
-        show_unquoted(io, ex, ind, -1)
+        show_unquoted(io, ex, ind, -1, quote_level)
     end
     print(io, '\n', " "^indent)
 end
-show_block(io::IO,head,    block,i::Int) = show_block(io,head, [], block,i)
-function show_block(io::IO, head, arg, block, i::Int)
+show_block(io::IO,head,    block,i::Int, quote_level::Int) = show_block(io,head, [], block,i, quote_level)
+function show_block(io::IO, head, arg, block, i::Int, quote_level::Int)
     if is_expr(arg, :block)
-        show_block(io, head, arg.args, block, i)
+        show_block(io, head, arg.args, block, i, quote_level)
+    elseif is_expr(arg, :quote)
+        show_block(io, head, arg.args, block, i, quote_level+0)
```

Comment:
`quote_level+0`?

Reply:
Originally it was (wrongly) `quote_level+1`, I've changed it to `quote_level+0` during debug and then i've forgot to commit the change where I've reunited that elseif branch with the previus one.